go/common/l1_types.go

```diff
@@ -107,13 +115,23 @@ func (h *Header) Hash() L1RootHash {
 //}
 
 // rlpHash encodes x and hashes the encoded bytes.
-func rlpHash(x interface{}) (h L1RootHash) {
+func rlpHash(value interface{}) (L1RootHash, error) {
```

Comment:
this function was taken from go-ethereum unchanged.
I liked the fact that it didn't return an error, as it made the calling code simpler
It would be good to understand what happens in case of an encoding error. Could that even happen given the data structures used?
I would guess it would return the hash of an empty byte array.
If the encoding fails here, it will definitely fail when trying to publish this rollup or transaction.

I'd say we leave it without an error for simplicity.

Reply:
Agreed.
I changed this as the linter was complaining and I don't like to silence errors.
But as far as I can tell we're supplying the encoder so we shouldn't see any errors. On top of that, existing errors the function will return, as you mentioned, the hash of an empty byte array, which will error upstream.

Reply:
we should definitely print some warnings though

Reply:
I'll compromise with a log and a TODO. 
Once we get a logging system in place we'll just hook that up 👍